.github/workflows/desktop-ci.yaml

```diff
@@ -14,63 +13,60 @@ jobs:
-          - platform: "macos-latest" # for Intel based macs.
-            arch: "x86_64-apple-darwin"
-            args: "--target x86_64-apple-darwin"
-          - platform: "ubuntu-22.04" # for Tauri v1 you could replace this with ubuntu-20.04.
-            arch: ""
-            args: ""
-          - platform: "windows-latest"
-            arch: ""
-            args: ""
+          - platform: macos-latest
+            targets: aarch64-apple-darwin,x86_64-apple-darwin
+            builds:
+              - target: aarch64-apple-darwin
+                artifact_path: ./target/aarch64-apple-darwin/release/bundle/
+              - target: x86_64-apple-darwin
+                artifact_path: ./target/x86_64-apple-darwin/release/bundle/
+          - platform: ubuntu-22.04
+            targets: ''
+            builds:
+              - target: ''
+                artifact_path: ./target/release/bundle/
+          - platform: windows-latest
+            targets: ''
+            builds:
+              - target: ''
+                artifact_path: ./target/release/bundle/
 
     runs-on: ${{ matrix.platform }}
     steps:
       - uses: actions/checkout@v4
 
-      - name: setup node
+      - name: Setup Node
         uses: actions/setup-node@v4
         with:
           node-version: lts/*
 
-      - name: install Rust stable
+      - name: Install Rust stable
         uses: dtolnay/rust-toolchain@stable
         with:
-          # Those targets are only used on macOS runners so it's in an `if` to slightly speed up windows and Linux builds.
-          targets: ${{ matrix.platform == 'macos-latest' && 'aarch64-apple-darwin,x86_64-apple-darwin' || '' }}
+          targets: ${{ matrix.targets }}
 
-      - name: install dependencies (ubuntu only)
-        if: matrix.platform == 'ubuntu-22.04' # This must match the platform value defined above.
+      - name: Install dependencies (Ubuntu only)
+        if: matrix.platform == 'ubuntu-22.04'
         run: |
           sudo apt-get update
           sudo apt-get install -y libwebkit2gtk-4.0-dev libwebkit2gtk-4.1-dev libappindicator3-dev librsvg2-dev patchelf
 
-      - name: install frontend dependencies
-        run: npm install --force # change this to npm, pnpm or bun depending on which one you use.
+      - name: Install frontend dependencies
+        run: npm install --force
         working-directory: ./desktop-app
 
-      - name: build tauri app
+      - name: Build Tauri apps
         uses: tauri-apps/tauri-action@v0
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          args: ${{ matrix.args }}
+          args: ${{ matrix.builds[0].target != '' && format('--target {0}', matrix.builds[0].target) || '' }}
           projectPath: "./desktop-app"
 
-      - name: upload artifact ${{ matrix.platform }}
-        if: matrix.platform != 'macos-latest' # This must match the platform value defined above.
+      - name: Upload artifacts
```

Comment:
we should ony add artifacts on tags

Reply:
or do we want to have artifacts for every build?

Reply:
okay, let's keep it for now, maybe later we can create some maunal flow for creating preview